TodoList/Tests/TodoList.Api.Tests/Controllers/ListItemsControllerTest.cs

```diff
@@ -1,166 +1,178 @@
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            ListItem actualListItem;
+            responseMessage.TryGetContentValue(out actualListItem);
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(actualListItem, Is.EqualTo(expectedListItem).UsingListItemComparer());
+        }
+
+        [Test]
+        public void GetAsync_ById_ReturnsCorrectResponse()
+        {
+            var expectedListItem = new ListItem { Id = _guidOfFirstItem, Text = "text" };
+            _repository.GetAsync(Guid.Empty).Returns(expectedListItem);
+
+            var actionResult = _controller.GetAsync(Guid.Empty).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            ListItem actualListItem;
+            responseMessage.TryGetContentValue(out actualListItem);
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(actualListItem, Is.EqualTo(expectedListItem).UsingListItemComparer());
+        }
+
+        [Test]
+        public void GetAsync_WithoutParams_ReturnsTestItems_ReturnsCorrectResponse()
+        {
+            var expectedListItems = new List<ListItem>
+            {
+                new ListItem { Id = _guidOfFirstItem, Text = "text" },
+                new ListItem { Id = _guidOfSecondItem, Text = "giraffe" },
+                new ListItem { Id = _guidOfThirdItem, Text = "updated" }
+            };
+            _repository.GetAllAsync().Returns(expectedListItems);
+
+            var actionResult = _controller.GetAsync().Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            List<ListItem> actualListItems;
+            responseMessage.TryGetContentValue(out actualListItems);
+            
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(actualListItems, Is.EqualTo(expectedListItems).UsingListItemComparer());
+        }
+
+        [TestCase("")]
+        [TestCase("    ")]
+        [TestCase(null)]
+        public void PostAsync_ItemWithNoText_ReturnsCorrectErrorResponse(string postedText)
+        {
+            var expectedKeys = new[] { "Text" };
+            var postedListItem = new ListItem { Id = Guid.Empty, Text = postedText };
+
+            var actionResult = _controller.PostAsync(postedListItem).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            var actualLocation = responseMessage.Headers.Location;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(error.ModelState.Keys, Is.EqualTo(expectedKeys));
+            Assert.That(actualLocation, Is.EqualTo(null));
```

Comment:
This actually looks like forget temporary fix. There is much better, more descriptive, way to test for null – simply (assert that <something>) `Is.Null`) – fix all occurences, of course.